### PR TITLE
HEXONET: Support for long TXT records

### PR DESCRIPTION
**Pull request by buraktamturk**
This PR fixes the issue described in #1264

I haven't run the integration tests on the Hexonet provider (I guess it will be run when I create this PR?). However tested the provider manually with following cases:

```
    TXT("dnscontrol-multi", ["test1", "test2", "test2"]),
    TXT("dnscontrol-single", "test1"),
    TXT("dnscontrol-long", "kRYDB7KBOeRRye0LwHnaHuvLiP1XsnJRUbU4cYPZeG1YyIDlKTEm6lijzw5ZDTcELICSKf7k1gTTCPT3KDCgqPRGccdKicqAv35JsbpklCxSafmZW5UosHtU9jZrfiwttTR9dGFVaNsnXpTBmb56A3my0aSki28sOwJFc0l2xj4P3VRTPOnWzBLLSSbMjCbdCvXSimUZUEI2d9Qxq19xmW2JoMLb4GXl5tXbjs1pvpUzs2VOYWlJOGLvBwU53IUqAG5G60TLx30OcKEF8qsNqYDqGVGPDNRYZjQFIloOB92y"),
```

Both preview and push command worked very well. I have checked the response from Hexonet DNS servers and everything seems okay.

```
;; ANSWER SECTION:
dnscontrol-multi.robinsoft.org. 300 IN  TXT     "test1" "test2" "test2"

;; ANSWER SECTION:
dnscontrol-single.robinsoft.org. 300 IN TXT     "test1"

;; ANSWER SECTION:
dnscontrol-long.robinsoft.org. 300 IN   TXT     "kRYDB7KBOeRRye0LwHnaHuvLiP1XsnJRUbU4cYPZeG1YyIDlKTEm6lijzw5ZDTcELICSKf7k1gTTCPT3KDCgqPRGccdKicqAv35JsbpklCxSafmZW5UosHtU9jZrfiwttTR9dGFVaNsnXpTBmb56A3my0aSki28sOwJFc0l2xj4P3VRTPOnWzBLLSSbMjCbdCvXSimUZUEI2d9Qxq19xmW2JoMLb4GXl5tXbjs1pvpUzs2VOYWlJOGLvBwU53IU" "qAG5G60TLx30OcKEF8qsNqYDqGVGPDNRYZjQFIloOB92y"
````

### CODE REVIEW

**Comment by tlimoncelli** [comment]
To run the integration tests, just do:

```
cd dnscontrol/integrationTest
export HEXONET_DOMAIN='a-b-c-movies.com'
export HEXONET_ENTITY='OTE'
export HEXONET_PW='test.passw0rd'
export HEXONET_UID='test.user'
go test -v -verbose -provider HEXONET
```

You can run specific tests via the start/end flags:

```
go test -v -verbose -provider HEXONET -start 11 -end 12
```

(Those creds are publicly advertised. This is not a security problem as they only work in OTE.)

File: providers/hexonet/auditrecords.go [hunk 1]
```diff
@@ -9,16 +9,6 @@ import (
...
 
-	// if err := recordaudit.TxtNoLongStrings(records); err != nil {
-	// 	return err
-	// }
-	// Not needed as of 2021-03-26
-
-	if err := recordaudit.TxtNoMultipleStrings(records); err != nil {
-		return err
-	}
-	// Still needed as of 2021-03-07
```
Comment (tlimoncelli) [inline providers/hexonet/auditrecords.go:20]: While your in this file, could you update those dates to be `2021-10-03`?  Thanks!

**Comment by tlimoncelli** [comment]
This all tests out for me too.  Just change those dates and I'll merge.

Thanks for fixing this!

**Comment by buraktamturk** [comment]
I just did! Thank you.

**Comment by papakai** [comment]
@buraktamturk many thanks for contributing - lack of time on my end, very appreciated! Let me forward my deepest gratitude for caring.

**Comment by tlimoncelli** [comment]
The unit tests were failing.  I fixed it and also solved the long-standing problem of TXT records not working if the last byte was whitespace.

The new PR is https://github.com/StackExchange/dnscontrol/pull/1283 I'll merge that one and close this one.

(Sadly this will show me as the author, not @buraktamturk , which I apologize for in advance.  I really need to learn git better so that doesn't happen.)

**Pull request closed without merging**